Homework 3/src/SortedLinkedList.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * Implements a Linked List data structure that is always sorted and uses only recursion.
+ * Andrew Id:
+ *
+ * @author
+ */
+public class SortedLinkedList implements MyListInterface {
+
+    private class Node {
+        String data;
+        Node next;
+
+        private Node() {
+            data = null;
+            next = null;
+        }
+    }
+
+    private Node head;
+
+    public SortedLinkedList(String[] array) {
+        if (array != null && array.length > 0)
+            constructorHelper(array, 0);
+    }
+
+    private void constructorHelper(String[] array, int currentIndex) {
+            this.add(array[currentIndex++]);
+            if (currentIndex < array.length)
+                constructorHelper(array, currentIndex);
+        }
+
+    /**
+     * Inserts a new String.
+     * Do not throw exceptions if invalid word is added (Gently ignore it).
+     * No duplicates allowed and maintain the order in ascending order.
+     * @param value String to be added.
+     */
+    @Override
+    public void add(String value) {
+        if (value == null || value.equals("")) return;
+
+        if (head == null) {
+            head = new Node();
+            head.data = value;
+            return;
+        }
+
+        addHelper(head, value, null);
```

Comment:
Why do you need to pass and an Instance variable to a method? It should be available to the method automatically.

Reply:
It's only on the first call that I pass head. The function is recursive.